src/to-do-List/components/to-do-list.tsx

```diff
@@ -0,0 +1,104 @@
+import { CSSProperties, useState } from "react";
+import { TaskTextProperties } from "../typings/task-text-properties";
+import { Link } from "react-router-dom";
+import { SpanProperties } from "../typings/span-task-properties";
+//funcion que recibe spanProperties con la const spanStyle que dará estilo al elemento span
+function spanTask(properties: SpanProperties) {
```

Comment:
Create custom component instead of a function that return the styles:

```tsx
import { PropsWithChildren } from "react";

type TaskTextProperties = {
  // ...
  onClick: () => void
}

function TaskText(properties: PropsWithChildren<TaskTextProperties>) {
  const spanStyle = // ...

  return (
    <span className="spanTask" style={spanStyle} onClick={properties.onClick}>
      {properties.children}
    </span>
  )
}
```

And then:

```tsx
<TaskText onClick={handleDelete}>x</TaskText>
<TaskText marginLeft="15px"  onClick={handleCopy}>copy</TaskText>
```